src/index.ts

```diff
@@ -150,6 +150,34 @@ const getAssetFromKV = async (event: FetchEvent, options?: Partial<Options>): Pr
     }
   })()
 
+  // formats the etag depending on the response context. if the entityId
+  // is invalid, returns an empty string (instead of null) to prevent the
+  // the potentially disastrous scenario where the value of the Etag resp
+  // header is "null". Could be modified in future to base64 encode etc
+  const formatETag = (entityId: any = pathKey, validatorType: string = 'strong') => {
+    if (!entityId) {
+      return ''
+    }
+    switch (validatorType) {
+      case 'weak':
+        if (!entityId.startsWith('W/')) {
+          return `W/${entityId}`
```

Comment:
BTW. The format of Weak ETag is `W/"xxxxxxx"`. Should `formatETag` validates quote before adding prefix `W/` ?

Reply:
I tested this actually: if the double quotes are omitted on strong etag, the weak validator is still respected by Cloudflare. So if the strong etag for some reason isn't bounded by quoted strings, the weak version of it should not enforce quoted strings either. In other words:
• The weak etag should only add `W/`
• The strong etag should add the quoted-strings
```
# compression performed, etag retained
etag: xyc
etag: W/xyc

# compression performed, etag retained
etag: "xyc"
etag: W/"xyc"
```